app/helpers/check_answers_helper.rb

```diff
@@ -0,0 +1,22 @@
+module CheckAnswersHelper
+  def get_answered_questions_total(subsection_pages, case_log)
+    questions = subsection_pages.values.map do |page|
+      page["questions"].keys
+    end
+    return questions.count {|questions_for_page| questions_for_page.none? { |question| case_log[question].blank?}}
```

Comment:
How about

```suggestion
   questions = subsection_pages.values.flat_map do |page| 
       page["questions"].keys 
    end
   questions.count { |question| case_log[question].present? }
```

Reply:
Hmm, that fails the tests, is it meant to be just a refactor or a different logic ?

Reply:
Different logic I think.

describe "Get answered questions total" do

it "returns 0 if only 1 question on a page gets answered" do

Doesn't make sense to me? I'd expect this to return 1 if 1 question has been answered but maybe I'm missing what this is supposed to do?

Reply:
So from what I understand, we've got check_answers page which displays a list of pages.

The 'number of questions' wording is confusing, because I think we're actually trying to say how many pages out of all in subsection have been filled in. So if one page has only half the questions completed this method would return 0.

Is that not how it's meant to work?

Reply:
Ignore that, we're just chatting about it now